### PR TITLE
chore: support ingestion ecs fargate

### PR DESCRIPTION
**Pull request by dengmingtong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

support ingestion ecs fargate

## Implementation highlights

Support ingestion ecs fargate
1. Add IngestionServerStackV2 for fargate.
2. Create resources according condition for 
- HTTP/HTTPS, 
- enable/disable alb access log
- enable/disable auth
- enable/disable aga
3. Add UT for V2 stack
4. Ingestion test for below cases
- S3 v2 enable all
- S3 v2 disable all
- KDS v2 enable all
- KDS v2 disable all
- MSK v2 enable all
- MSK v2 disable all
- S3 v1 test.

## Test checklist

- [x] add new test cases
- [x] all code changes are covered by unit tests
- [x] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [x] deploy ingestion server
    - [x] with MSK sink
    - [x] with KDS sink
    - [x] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [x] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Review by zxkane — COMMENTED** [review]
I'll play with it more.

Another suggestion is to use "Construct" as a style for managing resources in CDK code, rather than using helper methods.

File: src/ingestion-server/server/parameter.ts [hunk 1]
```diff
@@ -215,6 +215,14 @@ export function createStackParameters(scope: Construct, props: {deliverToKinesis
     minValue: 60,
   });
 
+  const fargateWorkerStopTimeoutParam = new CfnParameter(scope, 'FargateWorkerStopTimeout', {
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/ingestion-server/server/parameter.ts:218]: The fargate service container stopTimeout max value is 120 seconds. But EC2 does not have this limitation, our existing WorkerStopTimeout for EC2 also does not have this limitation, it's default value is 330.

File: .github/workflows/dist-scan.yml [hunk 2]
```diff
@@ -56,10 +56,10 @@ jobs:
         run: |
           cfn-lint --version
           TEMPLATE_ROOT=templates/default
-          cfn-lint -i W3005 W2001 W3045 -e -r us-east-1,ap-northeast-1 -t $TEMPLATE_ROOT/*.template.json
-          cfn-lint -i W3005 W2001 W3045 -e -r ap-east-1 --ignore-templates $TEMPLATE_ROOT/data-reporting-quicksight-stack.template.json --ignore-templates $TEMPLATE_ROOT/*NewServerlessRedshift*.nested.template.json --ignore-templates $TEMPLATE_ROOT/data-pipeline-stack.template.json --ignore-templates $TEMPLATE_ROOT/datapipeline*.nested.template.json --ignore-templates $TEMPLATE_ROOT/cloudfront-s3-control-plane-stack-global* --ignore-templates $TEMPLATE_ROOT/*cognito-control-plane-stack.template.json --ignore-templates $TEMPLATE_ROOT/public-exist-vpc-custom-domain-control-plane-stack.template.json -t $TEMPLATE_ROOT/*.template.json
+          cfn-lint -i W3005 W2001 W3045 W8003 -e -r us-east-1,ap-northeast-1 -t $TEMPLATE_ROOT/*.template.json
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply .github/workflows/dist-scan.yml:59]: W8003 is warning for "Fn::Equals element will alway return true",

File: src/ingestion-server-v2-stack.ts [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+}
+
+export class IngestionServerV2NestedStack extends NestedStack {
+  constructor(
+    scope: Construct,
+    id: string,
+    props: IngestionServerV2NestStackProps,
+  ) {
+    super(scope, id, props);
+    const featureName = 'IngestionServer ' + id;
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/ingestion-server-v2-stack.ts]: Added

File: src/ingestion-server/custom-resource/update-alb-rules-v2.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
+/**
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/ingestion-server/custom-resource/update-alb-rules-v2.ts]: refactored existing update-alb-rules.ts file and use condition to check auth is enable or disable.

File: src/ingestion-server-v2-stack.ts [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+
+    this.templateOptions.description = `(${SolutionInfo.SOLUTION_ID}-ing) ${SolutionInfo.SOLUTION_NAME} - ${featureName} ${SolutionInfo.SOLUTION_VERSION_DETAIL}`;
+
+    // Vpc
+    const vpc = getExistVpc(this, 'from-vpc', {
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/ingestion-server-v2-stack.ts]: Refactored to share code for v1 and v2

File: src/ingestion-server-v2-stack.ts [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+import {
+  FleetV2Props,
+  IngestionServerV2,
+  IngestionServerV2Props,
+  S3SinkConfig,
+  KinesisSinkConfig,
+  KafkaSinkConfig,
+} from './ingestion-server/server-v2/ingestion-server-v2';
+
+interface IngestionServerV2NestStackProps extends StackProps {
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/ingestion-server-v2-stack.ts]: Refactored to share with previous version.

File: src/control-plane/backend/lambda/api/config/dictionary.json [hunk 7]
```diff
@@ -15,6 +15,9 @@
       "Ingestion_s3": "ingestion-server-s3-stack.template.json",
       "Ingestion_kafka": "ingestion-server-kafka-stack.template.json",
       "Ingestion_kinesis": "ingestion-server-kinesis-stack.template.json",
+      "Ingestion_s3_Fargate": "ingestion-server-v2-s3-stack.template.json",
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/control-plane/backend/lambda/api/config/dictionary.json:18]: Refactored to use a single template for V2 for all sink type.

**Comment by dengmingtong** [comment]
> I'll play with it more.
> 
> Another suggestion is to use "Construct" as a style for managing resources in CDK code, rather than using helper methods.

Already refactor code to use Construct to replace methods

File: src/ingestion-server/server/parameter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/ingestion-server/server/parameter.ts]: refactored code to use WorkerStopTimeout for v1 and v2.

**Review by chenhaiyun — APPROVED** [review]

**Pull request merged**